### ISSUE TITLE
False positive "no-member" on module.__version__ with pylint 2.15

### ISSUE THREAD

**Issue by andy-maier** [PyCQA/pylint#7919]
### Bug description

With the following script:
```
$ cat zhmc_version.py 
"""
Pylint issue: False positive "no-member" on using zhmcclient. __version__
"""

import zhmcclient

print(zhmcclient.__version__)
```

pylint reports this:

```
$ pylint zhmc_version.py 
************* Module zhmc_version
zhmc_version.py:7:6: E1101: Module 'zhmcclient' has no '__version__' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

However, the `__version__` member exists, as shown when executing the script:
```
$ python zhmc_version.py 
1.6.0.dev1
```

Repo: https://github.com/zhmcclient/python-zhmcclient

### Configuration

```ini
# no config file
```

### Command used

```shell
1. `pip install zhmcclient`
2. `pylint zhmc_version.py` - with file shown above
```

### Pylint output

```shell
************* Module zhmc_version
zhmc_version.py:7:6: E1101: Module 'zhmcclient' has no '__version__' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

pylint does not report any issue for this source file.

### Pylint version

```shell
pylint 2.15.8
astroid 2.12.13
Python 3.9.15 (main, Oct 11 2022, 22:26:39) 
[Clang 13.0.0 (clang-1300.0.29.30)]
```

### OS / Environment

MacOS

**Comment by clavedeluna** [comment]
What @GideonBear said is 💯 correct. You can test this out by cloning the repo and changing the `__init__.py` file from `from ._version import *       # noqa: F401` to `from ._version import __version__` then the issue disappears if you test it out by installing the local repo.

Maintainers, is this an issue to close as won't fix or to keep as an example that should work if `*` imports are improved in the future?

**Comment by andy-maier** [comment]
Given that "close as won't fix" was brought up as one option to deal with this, I would like to mention that the approach of using `from ._{submodule} import *` in `__init__.py`with a (carefully curated) list of exported names in the submodules via `__all__` is an efficient way of having code in submodules of a package, but having the symbols for users all in the package namespace in order not to bother the users with the submodule structure.

The alternative to import every symbol exported in a submodule into `__init__.py` is cumbersome and error-prone, and I would rather globally disable the pylint no-member check if you decide not to fix this.

Note that we are serious users of pylint: We let our builds fail if pylint reports anything, in many of our projects (not all yet, but we're getting there).

Also note that there is a new issue with the no_member check, reported in issue #7972, this time on the standard Python `io` and `os` packages.

Given the behavior of the new issue in which the false positive only happens on Python 3.11 and only with pylint<2.15, I re-checked the error reported in this issue here with the same Python and pylint versions. The "issue" column indicates where the no-member error is reported for the `__version__` attribute:

```
Python      pylint     astroid    issue
-----------------------------------------

3.10.9      2.10.0     2.7.2      Yes
3.10.9      2.13.0     2.11.7     Yes
3.10.9      2.14.0     2.11.7     Yes
3.10.9      2.15.0     2.12.13    Yes

3.11.0      2.13.0     2.11.7     Yes
3.11.0      2.14.0     2.11.7     Yes
3.11.0      2.15.0     2.12.13    Yes
```

**Comment by Pierre-Sassoulas** [comment]
It's a false positive for an error level message which I would consider one of the more problematic bug we can have (just a little under outright crashing). But on another hand I think I remember that mypy > 0.990 started to fail for this kind of use cases (?) and wildcard import are discouraged.

**Comment by mbyrnepr2** [comment]
The issue seems to be a combination of using wildcard import when `__all__` is in the other module; If `__version__` does not begin with an underscore then the false positive goes away. On the other hand, if `__all__` is not defined, then `__version__` cannot be imported by Python because it begins with an underscore.

**Comment by mbyrnepr2** [comment]
[Discussion related to previous comment in astroid](https://github.com/PyCQA/astroid/issues/1362).
The `wildcard_import_names` function is defined but doesn't seem to be used anymore.

**Comment by mbyrnepr2** [comment]
I’ll close this because it is a duplicate of the astroid issue and there is already a discussion about the root cause there. Please re-open if needed (this or the astroid issue).
